src/main.py

```diff
@@ -18,28 +20,97 @@ async def on_ready():
     print(client.user.id)
 
 @client.event
-async def on_reaction_add(reaction, user):
-    if reaction.message.channel.id != VERIFY_CHAN or client.user != reaction.message.author:
-        return
+async def on_reaction_add(reaction: discord.Reaction, user: discord.User):
+    # Ignore messages that have been handled previously
     if reaction.message.id in leah.posted:
         return
-    leah.posted.add(reaction.message.id)
-    txt = reaction.message.content.split('\n')
-    other = reaction.message.mentions[0]
-    embed = discord.Embed(title=f"Some amazing art by {str(other)}", type="rich", color=other.color, description=txt[-2])
-    embed.set_image(url=txt[-1])
-    gallery = client.get_channel(GALLERY_CHAN)
-    await gallery.send(embed=embed)
+
+    # Staff reactions to verification channel posts by this client
+    if reaction.message.channel.id == VERIFY_CHAN:
+        # Ignore reactions to posts not by this client in verification channel
+        if client.user != reaction.message.author:
+            return
+
+        # Publish verified posts to gallery channel
+        await publish_art(message=reaction.message)
+
+    # User reactions to posts in self-curated channel
+    elif reaction.message.channel.id in MOD_CHANS:
+        # Ignore reactions from users without any of the required roles
+        if len(SHOWCASE_ROLES) == 0 or len([r for r in user.roles if r.id in SHOWCASE_ROLES]) == 0:
+            return
+
+        # Ignore reactions from users other than the message author
+        if user != reaction.message.author:
```

Comment:
The vast majority of the reactions it will detect won't be users reacting to themselves, so I think we should do this before we bother checking the roles.

Reply:
Yep, noticed this today — aimply swapping the user and role checks around might suffice